### ISSUE TITLE
Certain Evernote notes fail to export

### ISSUE THREAD

**Issue by upnix** [akosbalasko/yarle#24]
With the most recent source (as of this post), certain notes in enex format will fail to export. I have attached an example note, which produces the following output:
```
chris yarle-master $ npm run start -- --enexFile=ShadowWork.enex --outputDir=./ShadowWork

> yarle-evernote-to-md@2.4.0 start /home/chris/yarle-master
> node ./dist/dropTheRope.js "--enexFile=ShadowWork.enex" "--outputDir=./ShadowWork"

/home/chris/yarle-master/dist/dropTheRope.js:21
    if (options.enexFile.endsWith('.enex')) {
                         ^

TypeError: Cannot read property 'endsWith' of undefined
    at Object.exports.run (/home/chris/yarle-master/dist/dropTheRope.js:21:26)
    at Object.<anonymous> (/home/chris/yarle-master/dist/dropTheRope.js:38:9)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! yarle-evernote-to-md@2.4.0 start: `node ./dist/dropTheRope.js "--enexFile=ShadowWork.enex" "--outputDir=./ShadowWork"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the yarle-evernote-to-md@2.4.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/chris/.npm/_logs/2020-05-26T16_40_58_055Z-debug.log
```
I will attach the debug log as well.

I had actually intended to report a different error, but upon updating my source I started getting this error. Once this is fixed, I suspect you will encounter a second problem exporting this note. The output of that error, from older source, was:
```
chris yarle-master $ npm run start -- --enexFile=ShadowWork.enex --outputDir=./ShadowWork

> yarle-evernote-to-md@2.3.3 start /home/chris/yarle-master
> node ./dist/dropTheRope.js "--enexFile=ShadowWork.enex" "--outputDir=./ShadowWork"

/home/chris/yarle-master/dist/process-resources.js:29
    const src = `./_resources/${relativeResourceWorkDir}/${resourceHashes[hash].replace(/ /g, '\ ')}`;
                                                                                ^

TypeError: Cannot read property 'replace' of undefined
    at addMediaReference (/home/chris/yarle-master/dist/process-resources.js:29:81)
    at Object.exports.processResources (/home/chris/yarle-master/dist/process-resources.js:24:26)
    at Object.exports.processNode (/home/chris/yarle-master/dist/process-node.js:19:39)
    at Object.exports.dropTheRope (/home/chris/yarle-master/dist/yarle.js:35:28)
    at Object.exports.run (/home/chris/yarle-master/dist/dropTheRope.js:20:11)
    at Object.<anonymous> (/home/chris/yarle-master/dist/dropTheRope.js:23:9)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! yarle-evernote-to-md@2.3.3 start: `node ./dist/dropTheRope.js "--enexFile=ShadowWork.enex" "--outputDir=./ShadowWork"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the yarle-evernote-to-md@2.3.3 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/chris/.npm/_logs/2020-05-26T16_08_04_816Z-debug.log
```
[2020-05-26T16_40_58_055Z-debug.log](https://github.com/akosbalasko/yarle/files/4683677/2020-05-26T16_40_58_055Z-debug.log)

[ShadowWork.enex.zip](https://github.com/akosbalasko/yarle/files/4683681/ShadowWork.enex.zip)

**Comment by akosbalasko** [comment]
Hi @upnix ,

Thank you for reporting this, it looks like a bug. I'll take a look at it, and keep you updated.

**Comment by akosbalasko** [comment]
@upnix I released a new version (2.4.1). As a hotfix, I skip internal images, screenshots as references to the generated Markdown file. Now as I see, your enex file can be transformed, please confirm. 
Nevertheless, I keep investigating this use-case to be able to support internal resources as well. 
Thank you for raising this up!

**Comment by John-Rule-Bose** [comment]
First off, thanks for creating this enex exporter, which I saw on the Obsidian forum.  I've just run it on a large enex file, and it failed on the 2nd note with the following error:

C:\code\yarle\dist\process-resources.js:45
    const timeStamp = resource['resource-attributes']['timestamp'];
                                                     ^

TypeError: Cannot read property 'timestamp' of undefined
    at processResource (C:\code\yarle\dist\process-resources.js:45:54)
    at Object.exports.processResources (C:\code\yarle\dist\process-resources.js:15:79)
    at Object.exports.processNode (C:\code\yarle\dist\process-node.js:19:39)
    at Object.exports.dropTheRope (C:\code\yarle\dist\yarle.js:29:36)
    at Object.exports.run (C:\code\yarle\dist\dropTheRope.js:21:15)
    at Object.<anonymous> (C:\code\yarle\dist\dropTheRope.js:37:9)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! yarle-evernote-to-md@2.6.2 start: `node ./dist/dropTheRope.js "--enexSource=jarule_Evernote_export_20200531.enex" "--outputDir=./md_notes" "--include-metadata"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the yarle-evernote-to-md@2.6.2 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

**Comment by akosbalasko** [comment]
Hi @John-Rule-Bose , 

Thanks for giving it a try! 
Yeah it looks like that my test files not really cover all cases :) let me have a look at it, I'll keep you updated on the results.

**Comment by akosbalasko** [comment]
Hi @John-Rule-Bose ,

I released a version(2.6.3) which at least does not break in such cases, but I'm afraid it will ignore to store some attachment of your notes. Missing of 'resource-attributes' indicates that there is no filename  and timestamp stored for an attachment, which is tricky to be stored.
However I tried to test it, I was not able to reproduce this issue. Could you please have a hint about the type of your 2nd note? Is this a web-clip?

**Comment by John-Rule-Bose** [comment]
Here’s an export of just a single note which causes the problem.

-John

**Comment by akosbalasko** [comment]
I'm sorry John, I don't see anything here.

**Comment by John-Rule-Bose** [comment]
Meaning you didn’t receive an attachment?  Or it didn’t produce the error?  I can resend...

John

On May 31, 2020, at 4:56 PM, Akos Balasko <notifications@github.com> wrote:

﻿

I'm sorry John, I don't see anything here.

**Comment by akosbalasko** [comment]
No attachment. Please resend it directly to me: akos0215@gmail.com.

Thank you!

John-Rule-Bose <notifications@github.com> ezt írta (időpont: 2020. máj.
31., V, 23:09):

**Comment by akosbalasko** [comment]
A new version is released (2.6.5) that resolves the issue.
